lib/rpc/handler/FallbackHandler.ts

```diff
@@ -0,0 +1,165 @@
+        this.log.error(
+          `${alarmEvent.providerId} becomes UNHEALTHY due to ${alarmEvent.previousState}=>ALARM in ${alarmEvent.alarmName}`
+        )
+      }
+    } else if (alarmEvent.state === 'OK') {
+      healthinessUpdate = await this.updateDbItemWhenOk(alarmEvent)
+      if (healthinessUpdate.isChanged()) {
+        metric.putMetric(`RPC_GATEWAY_FALLBACK_${alarmEvent.providerId}_INTO_HEALTHY`, 1, MetricLoggerUnit.Count)
+        this.log.error(
+          `${alarmEvent.providerId} becomes HEALTHY due to ${alarmEvent.previousState}=>OK in ${alarmEvent.alarmName}`
+        )
+      }
+    }
+    return healthinessUpdate
+  }
+
+  private async updateDbItemWhenAlarm(alarmEvent: AlarmEvent): Promise<HealthinessUpdate> {
+    const state: ProviderHealthState | null = await this.healthStateRepository.read(alarmEvent.providerId)
+    if (state === null) {
+      const newState: ProviderHealthState = {
+        healthiness: ProviderHealthiness.UNHEALTHY,
+        ongoingAlarms: [alarmEvent.alarmName],
+        version: 1,
+      }
+      await this.healthStateRepository.write(alarmEvent.providerId, newState)
+    } else {
+      const newOngoingAlarms = state.ongoingAlarms
+      if (!state.ongoingAlarms.includes(alarmEvent.alarmName)) {
+        newOngoingAlarms.push(alarmEvent.alarmName)
+      }
+      const newState: ProviderHealthState = {
+        healthiness: ProviderHealthiness.UNHEALTHY,
+        ongoingAlarms: newOngoingAlarms,
+        version: state.version + 1,
+      }
+      await this.healthStateRepository.update(alarmEvent.providerId, newState)
+    }
+
+    // Return true if it becomes unhealthy from healthy.
+    return new HealthinessUpdate(
+      state === null || state.healthiness === ProviderHealthiness.HEALTHY
+        ? ProviderHealthiness.HEALTHY
+        : ProviderHealthiness.UNHEALTHY,
+      ProviderHealthiness.UNHEALTHY
+    )
+  }
+
+  private async updateDbItemWhenOk(alarmEvent: AlarmEvent): Promise<HealthinessUpdate> {
+    const state: ProviderHealthState | null = await this.healthStateRepository.read(alarmEvent.providerId)
+    if (state === null || state.healthiness === ProviderHealthiness.HEALTHY) {
+      return new HealthinessUpdate(ProviderHealthiness.HEALTHY, ProviderHealthiness.HEALTHY)
+    }
+    if (!state.ongoingAlarms.includes(alarmEvent.alarmName)) {
+      return new HealthinessUpdate(ProviderHealthiness.UNHEALTHY, ProviderHealthiness.UNHEALTHY)
+    }
+    const newOngoingAlarms = state.ongoingAlarms.filter((alarmName) => alarmName !== alarmEvent.alarmName)
+    const newState: ProviderHealthState = {
+      healthiness: newOngoingAlarms.length === 0 ? ProviderHealthiness.HEALTHY : ProviderHealthiness.UNHEALTHY,
+      ongoingAlarms: newOngoingAlarms,
+      version: state.version + 1,
+    }
+    await this.healthStateRepository.update(alarmEvent.providerId, newState)
+    return new HealthinessUpdate(
```

Comment:
strange here that we pass two status, we should follow the comment above around how the class is constrcucted. Also, not sure if you even need a class? You only use it once and you can compare at the time you are calling "isChanged" anyway with a simple comparator

Reply:
I expalined myself above we can keep it there